### PR TITLE
t/m/snap-quota-thread: make the test more robust against delays

### PR DESCRIPTION
**Pull request by Meulengracht**
The way the spawner snap works is by being a bit abusive and just trying to spawn X threads every delay - now this seems to be mostly fine, but from testing it more it seems that once in a while the delay it takes from changing the cgroup thread allowance for currently running processes to the moment when the spawner actually get to spawn additional thread can take a while, causing the test to timeout.

Avoid this instability all together by restarting the service, causing the change to go into immediate effect.

### CODE REVIEW

File: tests/main/snap-quota-thread/task.yaml [hunk 1]
```diff
@@ -27,6 +27,21 @@ execute: |
   # count of threads expected is 33 including the main thread.
   SPAWNER_MAX_THREADS=33
 
+  function wait_for_thread_usage() {
+    local threadCount=0
+    local runs=$1
+    local expected=$2
+
+    for _ in $(seq "${runs}"); do
+        threadCount="$(snap quota group-one | awk '/current:/,/threads:/{printf "%s",$2} END {print ""}')"
```
Comment (bboozzoo) [inline tests/main/snap-quota-thread/task.yaml:36]: I suppose this should work too without resorting to END trickery with newlines
```suggestion
        threadCount="$(snap quota group-one | awk '/current:/,/threads:/{if ($1 ~ "threads:") { print $2 } }'
```

**Review by bboozzoo — APPROVED** [review]
LGTM

**Review by ernestl — APPROVED** [review]
Thanks

File: tests/main/snap-quota-thread/task.yaml (hunk 1 above)
Comment (Meulengracht) [reply tests/main/snap-quota-thread/task.yaml:36]: nice! added

**Pull request merged**